tensorrt_llm/_torch/speculative/utils.py

```
@@ -39,6 +39,35 @@ def _is_effective_dynamic_tree(spec_config) -> bool:
+        import json
+        import os
+        cfg_path = os.path.join(str(draft_dir), "config.json")
+        if not os.path.isfile(cfg_path):
+            return target_vocab_size
+        with open(cfg_path) as f:
+            cfg = json.load(f)
+        vs = cfg.get("vocab_size")
+        if vs is None:
+            vs = (cfg.get("text_config") or {}).get("vocab_size")
+        return int(vs) if vs else target_vocab_size
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)